index.bs

```diff
@@ -147,35 +151,125 @@ Exactly how external changes are reflected in the data structures defined by thi
+|a|'s [=file system locator/path=] is [=the same path as=] |b|'s [=file system locator/path=].
+
+The <dfn export data-lt="locating an entry">locate an entry</dfn> algorithm takes a
+[=/file system locator=] |locator| and returns either a [=/file system entry=] or null.
+If |locator| is a [=file locator=], this method must return either a [=file entry=] or null.
+If |locator| is a [=directory locator=], this method must return either a [=directory entry=] or null.
+The steps of this algorithm are [=implementation-defined=].
+
+The <dfn export data-lt="getting the locator">get the locator</dfn> algorithm takes a
+[=/file system entry=] |entry| and returns a [=/file system locator=] |locator|.
+
+If |entry| is a [=file entry=], |locator| must be a [=file locator=].
+If |entry| is a [=directory entry=], |locator| must be a [=directory locator=].
+If [=locating an entry=] given [=/file system locator=] |locator| returns a
+[=/file system entry=] |entry|, then [=getting the locator=] of |entry|
+must return |locator| if no intermediate file system operations were run.
+Typically the steps are symmetric to the [=locate an entry=] algorithm.
+
+If |entry| is the root directory of the [=origin private file system=], then
+|entry|'s [=file system entry/name=] must be the empty string and
+[=getting the locator=] of |entry| must return a [=/file system locator=]
+whose [=file system locator/path=] is [=list/empty=].
+
+If [=locating an entry=] given [=/file system locator=] |locator| returns a
+[=/file system entry=] |entry| that is not the root directory of the
+[=origin private file system=], then |entry|'s [=file system entry/name=] must
+be the same [=string=] as the last [=list/item=] of
+|locator|'s [=file system locator/path=].
+
+A <dfn export>file system path</dfn> is a [=/list=] of one or more [=strings=].
+This may be a virtual path that is mapped to real location on disk or in memory,
+may correspond directly to a path on the local file system, or may not
+correspond to any file on disk at all. The actual physical location of the
+corresponding [=/file system entry=] is [=implementation-defined=].
+
+<p class=example id=example-path>Let |path| be the [=/list=]
+« "`data`", "`drafts`", "`example.txt`" ».
+There is no expectation that a file named `example.txt` exists anywhere on disk.
+
+A [=/file system path=] |a| is <dfn for="file system path">the same path as</dfn>
+a [=/file system path=] |b| if
+|a|'s [=list/size=] is the same as |b|'s [=list/size=] and
+[=list/for each=] |index| of |a|'s [=list/indices=]
+|a|.\[[|index|]] is |b|.\[[|index|]].
+
+Note: The contents of a [=/file system locator=], including its
+[=file system locator/path=], are not expected to be shared in their entirety
+with the website process. The [=/file system path=] might contain components
+which are not known to the website unless the [=/file system locator=] is later
+[=file system locator/resolved=] relative to a parent [=directory locator=].
```

Comment:
I think we want `<p class=warning>` for this. I think "website process" is probably okay language for now. Although maybe we should say "agent cluster" or "agent".

Reply:
Ack. Added the warning label, and happy to update the other text if desired